### PR TITLE
PP-500: Fix abysmal qrerun performance with large spool file staging

### PR DESCRIPTION
**Pull request by subhasisb**
#### Issue-ID

* **[PP-500](https://pbspro.atlassian.net/browse/PP-500)**

#### Problem description
* When a job with a large stdout/err file is rerun, the performance is very bad. 

#### Cause / Analysis
* For files as large as 200 GB, it results in 60k+ transactions from mom to server and the same number back to mom when the file is staged back to mom. This is because the block size being used is 4096 bytes which results in the large number of transactions. The overhead of processing this is also very large at the server. 

#### Solution description
* Since direct TCP connections are being used between the server and mom for these large spool file transfers, it is possible to increase the block size to much larger, ie, 64k. This reduces the number of transactions by 16 times.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/performance/pbs_rerunjob_file_transfer_perf.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid, max_attempts=30, interval=5)
+	now2 = int(time.time())
+	self.logger.info("Job %s took %d seconds to start\n", jid, (now2 - now1))
+
+        time.sleep(5)
+
+	now1 = int(time.time())
+	self.server.rerunjob(jid)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid, max_attempts=500, interval=5)
+	now2 = int(time.time())
```
Comment (prakashcv13) [inline test/tests/performance/pbs_rerunjob_file_transfer_perf.py]: an extra space on line 90?

File: test/tests/performance/pbs_rerunjob_file_transfer_perf.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+        j.create_script(test, hostname=self.server.client)
+
+	now1 = int(time.time())
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid, max_attempts=30, interval=5)
+	now2 = int(time.time())
+	self.logger.info("Job %s took %d seconds to start\n", jid, (now2 - now1))
+
+        time.sleep(5)
```
Comment (Klovely) [inline test/tests/performance/pbs_rerunjob_file_transfer_perf.py]: Just one query I have why we are using time.sleep(5)?
and if it is intentional, please add comment .

**Comment by ashwathraop** [comment]
Code looks fine.

**Comment by subhasisb** [comment]
@Klovely added the requested comment

**Comment by Klovely** [comment]
Thanks Subhasis.
Rest code looks fine to me .

**Comment by bhroam** [comment]
Looks good to me too.  I'll merge it in.

**Comment by subhasisb** [comment]
Ooops! I had not squashed the commits yet @bhroam

**Pull request merged**